bitnami/prometheus-operator/templates/prometheus/prometheus.yaml

```diff
@@ -212,7 +212,6 @@ spec:
       volumeMounts:
         - mountPath: /prometheus
           name: prometheus-{{ template "prometheus-operator.prometheus.fullname" . }}-db
-          subPath: prometheus-db
```

Comment:
shouldn't it be optional depending on `storage.disableMountSubPath`?

Reply:
Yeah.. This was based on the assumption that `storageSpec`'s options are currently only passed through as is, and with no documentation in particular. I can change it to be conditional instead.

That said, `storage.disableMountSubPath` is deprecated and will presumably be removed soon.

Reply:
Oh, fun. So it turns out that the Helm chart still ships an old version of the `Prometheus` CRD that doesn't even have the option (https://github.com/coreos/prometheus-operator/blob/v0.40.0/bundle.yaml#L9274, doesn't exist in https://github.com/bitnami/charts/blob/b0c0732b5ef07d31a84cb64d2770b34fee6ecc57/bitnami/prometheus-operator/crds/crd-prometheus.yaml). So there are no existing users to break, unless they manually installed a different CRD.

Reply:
So, I think the best approach would be, keep it optional depending on `storage.disableMountSubPath` at least until we had upgraded our CRD version, don't you think so?

Reply:
The current CRD version doesn't allow setting `storage.disableMountSubPath` anyway.

Reply:
I'm still not really convinced that this is the correct path, but 362d7c90d makes it configurable.